xonsh/history/json.py

```diff
@@ -379,20 +379,24 @@ def flush(self, at_exit=False):
         self.buffer.clear()
         return hf
 
-    def items(self):
+    def items(self, reverse=False):
         """Display history items of current session."""
-        for item, tss in zip(self.inps, self.tss):
+        if reverse:
+            items = zip(reversed(self.inps), reversed(self.tss))
+        else:
+            items = zip(self.inps, self.tss)
+        for item, tss in items:
             yield {'inp': item.rstrip(), 'ts': tss[0]}
```

Comment:
Can anyone see a more elegant way of doing this?

Reply:
I wouldn't think that was possible since zip is an iterator and could in principle be infinite. It would really have to be `reversed(list(items))` which isn't nice.

Reply:
Oh, sorry yeah, from itertools you can use `izip()`, but I think you run into the same problem with reverse.